src/main/java/com/synopsys/integration/jira/common/server/builder/IssueRequestModelFieldsBuilder.java

```diff
@@ -0,0 +1,174 @@
+    public static final String SUMMARY = "summary";
+    public static final String ISSUE_TYPE = "issuetype";
+    public static final String COMPONENTS = "components";
+    public static final String PROJECT = "project";
+    public static final String DESCRIPTION = "description";
+    public static final String REPORTER = "reporter";
+    public static final String FIX_VERSIONS = "fixVersions";
+    public static final String PRIORITY = "priority";
+    public static final String LABELS = "labels";
+    public static final String TIME_TRACKING = "timetracking";
+    public static final String SECURITY = "security";
+    public static final String ENVIRONMENT = "environment";
+    public static final String VERSIONS = "versions";
+    public static final String DUE_DATE = "duedate";
+    public static final String ASSIGNEE = "assignee";
+
+    private final Map<String, Object> issueFields;
+
+    public IssueRequestModelFieldsBuilder() {
+        this.issueFields = new HashMap<>();
+    }
+
+    public IssueRequestModelFieldsBuilder copyFields(IssueRequestModelFieldsBuilder original) {
+        Map<String, Object> originalFields = original.build();
+        issueFields.putAll(originalFields);
+        return this;
+    }
+
+    public Map<String, Object> build() {
+        return Collections.unmodifiableMap(issueFields);
+    }
+
+    public IssueRequestModelFieldsBuilder setSummary(String summary) {
+        issueFields.put(SUMMARY, summary);
+        return this;
+    }
+
+    public IssueRequestModelFieldsBuilder setIssueType(String issueTypeId) {
+        return setIdField(ISSUE_TYPE, issueTypeId);
+    }
+
+    public IssueRequestModelFieldsBuilder setComponents(Collection<String> componentIds) {
+        return setIdFields(COMPONENTS, componentIds);
+    }
+
+    public IssueRequestModelFieldsBuilder setProject(String projectId) {
+        return setIdField(PROJECT, projectId);
+    }
+
+    public IssueRequestModelFieldsBuilder setDescription(String description) {
+        issueFields.put(DESCRIPTION, description);
+        return this;
+    }
+
+    public IssueRequestModelFieldsBuilder setReporter(String reporterId) {
+        return setNameField(REPORTER, reporterId);
+    }
+
+    public IssueRequestModelFieldsBuilder setFixVersions(Collection<String> versionIds) {
+        return setIdFields(FIX_VERSIONS, versionIds);
+    }
+
+    public IssueRequestModelFieldsBuilder setPriority(String priorityId) {
+        return setIdField(PRIORITY, priorityId);
+    }
+
+    public IssueRequestModelFieldsBuilder setLabels(Collection<String> labels) {
+        issueFields.put(LABELS, labels);
+        return this;
+    }
+
+    public IssueRequestModelFieldsBuilder setTimeTracking(String remainingEstimate, String originalEstimate) {
+        Map<String, String> timeTrackingMap = new HashMap<>();
+        timeTrackingMap.put("remainingEstimate", remainingEstimate);
+        timeTrackingMap.put("originalEstimate", originalEstimate);
```

Comment:
Should these be in constants?

Reply:
Meh, they aren't top level fields, so it could go either way.